src/mongoose_client_api/mongoose_client_api_messages.erl

```diff
@@ -128,7 +128,14 @@ encode(Msg, Timestamp) ->
                                         <<"http://www.jivesoftware.com/xmlns/xmpp/properties">>),
 
     BodyTag = exml_query:path(Msg, [{element, <<"body">>}]),
-
+    Thread = exml_query:path(Msg, [{element, <<"thread">>}, cdata]), 
```

Comment:
I suppose this can be rewrite like
```erlang
Thread = exml_query:path(Msg, [{element, <<"thread">>}, cdata]), 
ThreadParentList = exml_query:path(Msg, [{element, <<"thread">>}, {attr, <<"parent">>}], undefined),
ThreadAndThreadParentList = case {Thread, ThreadParentList} of
    {undefined, undefined} -> 
        [];
    {Thread, ThreadParent} ->
        [{<<"thread">>, ThreadParent}, {<<"thread">>, Thread}]
end,
L = [{<<"from">>, exml_query:attr(Msg, <<"from">>)},
       {<<"to">>, exml_query:attr(Msg, <<"to">>)},
       {<<"id">>, exml_query:attr(Msg, <<"id">>)},
       {<<"body">>, exml_query:cdata(BodyTag)},
       {<<"thread">>, Thread},
       {<<"timestamp">>, Timestamp}] ++ ThreadAndThreadParentList,
...
```

Reply:
Ya, I will change it. Looks much clearer.

Can the `ExtensionList` be chain appended the same way?
'''
L = [
       ...
       {<<"timestamp">>, Timestamp}] ++ ThreadAndThreadParentList ++ ExtensionList,
'''

Reply:
since these values are interrelated and should either be present or absent, better if we will or apply them or will not apply them

Reply:
Agree. Mind if I ask why 
'''
ThreadParent = exml_query:path(Msg, [{element, <<"thread">>}, {attr, <<"parent">>}])
'''
will crash the test when thread is not present with 

'''
 ====== Reason:    {badarg,
                      [{erlang,binary_to_existing_atom,
                           [<<"parent">>,utf8],
                           [{error_info,#{module => erl_erts_errors}}]},
                       {rest_helper,'-mapfromlist/1-fun-0-',1,
'''

Reply:
I suppose cuz expected `binary` instead of `atom`.

Reply:
`ThreadParent = exml_query:path(Msg, [{element, <<"thread">>}, {attr, <<"parent">>}]),` should be returning `undefined` if <thread> element is not found. Was spending quite some time on thinking why it crashes, I tried path with default to `undefined`, `none` and `<<>>`  all crashes with badArgs. If i check Rest output if the attr is missing such as id, it will return string 'undefined'